### PR TITLE
Improved support for Android TV with Leanback launcher

### PR DESCRIPTION
**Pull request by cemrich**
Install a channel showing the available programs in a row in the default Android TV launcher (Leanback).

### CODE REVIEW

File: .idea/jarRepositories.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      <option name="id" value="jboss.community" />
+      <option name="name" value="JBoss Community repository" />
+      <option name="url" value="https://repository.jboss.org/nexus/content/repositories/public/" />
+    </remote-repository>
+    <remote-repository>
+      <option name="id" value="BintrayJCenter" />
+      <option name="name" value="BintrayJCenter" />
+      <option name="url" value="https://jcenter.bintray.com/" />
+    </remote-repository>
+    <remote-repository>
```
Comment (cemrich) [inline .idea/jarRepositories.xml:19]: Already defined in build.gradle

File: .idea/jarRepositories.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="RemoteRepositoriesConfiguration">
+    <remote-repository>
+      <option name="id" value="central" />
+      <option name="name" value="Maven Central repository" />
+      <option name="url" value="https://repo1.maven.org/maven2" />
```
Comment (cemrich) [inline .idea/jarRepositories.xml:7]: This should be `mavenCentral()` in build.gradle But is it needed for this change?

File: .idea/jarRepositories.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    </remote-repository>
+    <remote-repository>
```
Comment (cemrich) [inline .idea/jarRepositories.xml:9]: Is this needed?

File: .idea/jarRepositories.xml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (cemrich) [inline .idea/jarRepositories.xml:1]: I think this file is not needed and can be replaced by references in build.gradle

File: .idea/misc.xml [hunk 5]
```diff
@@ -39,7 +39,7 @@
       </value>
     </option>
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8 (1)" project-jdk-type="JavaSDK">
```
Comment (cemrich) [inline .idea/misc.xml:42]: Please exclude this change.

File: .idea/modules.xml [hunk 6]
```diff
@@ -2,8 +2,10 @@
 <project version="4">
```
Comment (cemrich) [inline .idea/modules.xml:2]: Please exclude this change.

File: app/src/main/java/de/christinecoenen/code/zapp/RunOnInstallReceiver.java [hunk 7]
```diff
@@ -0,0 +1,12 @@
+package de.christinecoenen.code.zapp;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+
+public class RunOnInstallReceiver extends BroadcastReceiver {
+	@Override
+	public void onReceive(Context context, Intent intent) {
+
```
Comment (cemrich) [inline app/src/main/java/de/christinecoenen/code/zapp/RunOnInstallReceiver.java:10]: This does nothing?

File: app/src/main/java/de/christinecoenen/code/zapp/app/livestream/ui/detail/ChannelDetailActivity.java [hunk 8]
```diff
@@ -45,7 +45,7 @@
 
 public class ChannelDetailActivity extends FullscreenActivity implements StreamPageFragment.Listener {
 
-	private static final String EXTRA_CHANNEL_ID = "de.christinecoenen.code.zapp.EXTRA_CHANNEL_ID";
+	public static final String EXTRA_CHANNEL_ID = "de.christinecoenen.code.zapp.EXTRA_CHANNEL_ID";
```
Comment (cemrich) [inline app/src/main/java/de/christinecoenen/code/zapp/app/livestream/ui/detail/ChannelDetailActivity.java:48]: Don't expose this. Use static `getStartIntent `instead.

File: .idea/jarRepositories.xml (hunk 4 above)
Comment (cemrich) [reply .idea/jarRepositories.xml:1]: Okay, I see now this is generated by the latest version of intelliJ. This file can safely be ignored.

**Comment by upost** [comment]
Hi!
I just checked my version on a real TV and found it (mostly) works.
1. I needed to start the app once to make the channels show up. This is propably related to the RunOnInstallReceiver not being implemented. I'll check that.
2. The app refused to start any stream because it was disabled on mobile network. The TV is connected using LAN. I suspect the check inside the app is not taking LAN into account. I'll check that.
3. In the channel overview screen, if you press the right button on the remote control, the list of Mediathek content shows up. I think that's not intuitive, it should be on the menu and only there.
I'll do some changes in the next days.
But, good thing last: It basically works :-)

**Comment by upost** [comment]
Bad news first:
I did not manage to use the vector drawables for the channels. Neither using file:/// nor content:/// or https:// Uris worked (creating PNGs from the drawables did work, but the launcher won't display them). Maybe you have another idea...
Also I'm not sure why AS created some extra XML files.

Good news:
I removed the obsolete Receiver and fixed the network check in a way that it now actually checks what the option says (mobile or not). This way, ethernet connection is treated as Wifi connection.

Check it out.

**Comment by cemrich** [comment]
@upost Nice work, thank you! I am working on releasing v3.5.0 with a new download functionality. I will check your changes as soon as this release is stable, which may take a while.

**Comment by cemrich** [comment]
@upost Quick status update: Zapp 3.5.0 introduced some bugs I need to fix first. Your pull request however is not forgotten :)

**Comment by cemrich** [comment]
I did not forget this pull request, but will most likely not be able to work on it the next few weeks or even months. 

However, version [3.6.1](https://github.com/mediathekview/zapp/releases/tag/v3.6.1) introduces better management for metered vs. unmetered networks. So LAN should now be recognized as unmetered and downloads and streams should work without tweaking.

**Comment by upost** [comment]
Just as a brief note: On my mother's TV (where I installed my version on) the WDR is missing in the channel list on the leanback launcher, but it is available when zapping around. I'm currently investigating what the cause could be.

Still no success using the vector images or even deriving bitmaps from them on the fly. Looks like the launcher supports bitmaps in the resources only :-(

**Pull request closed without merging**